File: win-prog/ProgWinVB/Buttons and Labels and Scrolls (Oh, My!)/CheckBoxWithLabel/CheckBoxWithLabel.vb
'--------------------------------------------------
' CheckBoxWithLabel.vb (c) 2002 by Charles Petzold
'--------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class CheckBoxWithLabel
    Inherits Form

    Private lbl As Label

    Shared Sub Main()
        Application.Run(New CheckBoxWithLabel())
    End Sub

    Sub New()
        Text = "CheckBox Demo with Label"

        Dim cyText As Integer = Font.Height
        Dim cxText As Integer = cyText \ 2
        Dim astrText() As String = {"Bold", "Italic", _
                                    "Underline", "Strikeout"}

        lbl = New Label()
        lbl.Parent = Me
        lbl.Text = Text & ": Sample Text"
        lbl.AutoSize = True

        Dim i As Integer
        For i = 0 To 3
            Dim chkbox As New CheckBox()
            chkbox.Parent = Me
            chkbox.Text = astrText(i)
            chkbox.Location = New Point(2 * cxText, _
                                       (4 + 3 * i) * cyText \ 2)
            chkbox.Size = New Size(12 * cxText, cyText)
            AddHandler chkbox.CheckedChanged, _
                                    AddressOf CheckBoxOnCheckedChanged
        Next i
    End Sub

    Private Sub CheckBoxOnCheckedChanged(ByVal obj As Object, _
                                         ByVal ea As EventArgs)
        Dim fs As FontStyle = 0
        Dim afs() As FontStyle = {FontStyle.Bold, FontStyle.Italic, _
                                  FontStyle.Underline, FontStyle.Strikeout}
        Dim i As Integer

        For i = 0 To 3
            If DirectCast(Controls(i + 1), CheckBox).Checked Then
                fs = fs Or afs(i)
            End If
        Next i
        lbl.Font = New Font(lbl.Font, fs)
    End Sub
End Class

File: win-prog/ProgWinCS/Buttons and Labels and Scrolls (Oh, My!)/CheckBoxWithLabel/CheckBoxWithLabel.cs
//------------------------------------------------
// CheckBoxWithLabel.cs © 2001 by Charles Petzold
//------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class CheckBoxWithLabel: Form
{
     Label label;

     public static void Main()
     {
          Application.Run(new CheckBoxWithLabel());
     }
     public CheckBoxWithLabel()
     {
          Text = "CheckBox Demo with Label";

          int      cyText   = Font.Height;
          int      cxText   = cyText / 2;
          string[] astrText = {"Bold", "Italic", "Underline", "Strikeout"};

          label = new Label();
          label.Parent   = this;
          label.Text     = Text + ": Sample Text";
          label.AutoSize = true;

          for (int i = 0; i < 4; i++)
          {
               CheckBox chkbox = new CheckBox();
               chkbox.Parent = this;
               chkbox.Text = astrText[i];
               chkbox.Location = new Point(2 * cxText, 
                                               (4 + 3 * i) * cyText / 2);
               chkbox.Size = new Size(12 * cxText, cyText);
               chkbox.CheckedChanged += 
                              new EventHandler(CheckBoxOnCheckedChanged);
          }
     }
     void CheckBoxOnCheckedChanged(object obj, EventArgs ea)
     {
          FontStyle   fs   = 0;
          FontStyle[] afs  = { FontStyle.Bold,      FontStyle.Italic, 
                               FontStyle.Underline, FontStyle.Strikeout };

          for (int i = 0; i < 4; i++)
               if (((CheckBox) Controls[i + 1]).Checked)
                    fs |= afs[i];

          label.Font = new Font(label.Font, fs);
     }
}
